client/src/canvas/components/Canvas.tsx

```diff
@@ -52,13 +62,23 @@ class Canvas extends Component<Props> {
     context!.scale(100, 100);
 
     this.canvasRef.current!.addEventListener('mousemove', (ev) => {
+      const { x, y } = this.getMousePos(this.canvasRef.current, ev); 
+      this.drawBorder(context, x, y); 
+    })
+
+    this.canvasRef.current!.addEventListener('mouseout', () => {
+      // this.props.onMouseOut();
```

Comment:
These most definitely need to be changed. Think about what happens when the user selects a pixel to display the color picker, and then moves the mouse to actually select their color. The pixel that they selected would be changed. Likewise, if they accidentally move out of the canvas, there is no reason their selection shouldn't be saved.